File: app/src/main/java/com/example/eccard/filmesfamosos/ui/moviedetail/MoviePageAdapter.java
package com.example.eccard.filmesfamosos.ui.moviedetail;

import android.content.Context;

import androidx.annotation.Nullable;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;

import com.example.eccard.filmesfamosos.R;

class MoviePageAdapter extends FragmentStatePagerAdapter {

    private static final int TOTAL_OF_FRGS = 3;
    private final Context context;

    public MoviePageAdapter(FragmentManager fm, Context context) {
        super(fm);
        this.context = context;
    }

    @Override
    public Fragment getItem(int position) {
        Fragment frg = null;

        switch (position) {
            case 0:
                frg = new FrgSummary();
                break;
            case 1:
                frg = new FrgTrailers();
                break;
            case 2 :
                frg = new FrgReviews();
            default:
                break;
        }

        return frg;
    }

    @Override
    public int getCount() {
        return TOTAL_OF_FRGS;
    }

    @Nullable
    @Override
    public CharSequence getPageTitle(int position) {
        CharSequence charSequence;

        switch (position){
            case 0:
                charSequence = context.getString(R.string.summary);
                break;
            case 1:
                charSequence = context.getString(R.string.videos);
                break;
            default:
                charSequence = context.getString(R.string.review);
                break;
        }

        return charSequence;
    }
}

File: app/src/main/java/com/example/eccard/filmesfamosos/ui/moviedetail/MoviePageAdapter.kt
package com.example.eccard.filmesfamosos.ui.moviedetail

import android.content.Context
import androidx.fragment.app.Fragment
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import com.example.eccard.filmesfamosos.R
import com.example.eccard.filmesfamosos.ui.moviedetail.reviews.FrgReviews
import com.example.eccard.filmesfamosos.ui.moviedetail.summary.FrgSummary
import com.example.eccard.filmesfamosos.ui.moviedetail.trailers.FrgTrailers

internal class MoviePageAdapter(fm: FragmentManager, private val context: Context) : FragmentStatePagerAdapter(fm) {

    override fun getItem(position: Int): Fragment? {
        var frg: Fragment? = null

        when (position) {
            0 -> frg = FrgSummary()
            1 -> frg = FrgTrailers()
            2 -> frg = FrgReviews()
            else -> {
            }
        }

        return frg
    }

    override fun getCount(): Int {
        return TOTAL_OF_FRGS
    }

    override fun getPageTitle(position: Int): CharSequence? {
        val charSequence: CharSequence

        when (position) {
            0 -> charSequence = context.getString(R.string.summary)
            1 -> charSequence = context.getString(R.string.videos)
            else -> charSequence = context.getString(R.string.review)
        }

        return charSequence
    }

    companion object {
        private val TOTAL_OF_FRGS = 3
    }
}